rest/replicator_test.go

```diff
@@ -3810,3 +3810,236 @@ func TestActiveReplicatorPullConflictReadWriteIntlProps(t *testing.T) {
+		return func() { require.NoError(t, ar.Stop(), "Error stopping replication") }
+	}
+
+	// requireBodyEmpty asserts that the specified document body is empty.
+	requireBodyEmpty := func(rt *RestTester, docID string) {
+		var body []byte
+		_, err := rt.Bucket().Get(docID, &body)
+		require.EqualError(t, err, fmt.Sprintf("Error during Get %s: key not found", docID))
+	}
+
+	// waitForRevision waits until the specified tombstone revision is available
+	// in the bucket backed by the specified RestTester instance.
+	waitForTombstone := func(rt *RestTester, docID, revID string) {
+		require.NoError(t, rt.WaitForCondition(func() bool {
+			doc, _ := rt.GetDatabase().GetDocument(docID, db.DocUnmarshalAll)
+			return doc.IsDeleted()
+		}))
+		doc, err := rt.GetDatabase().GetDocument(docID, db.DocUnmarshalAll)
+		require.NoError(t, err, "Error reading document from bucket")
+		require.Equal(t, revID, doc.SyncData.CurrentRev)
+		require.Empty(t, doc.Body(), "Document body should be empty")
+	}
+
+	// requireBodyEmpty asserts that the specified document body is not empty.
+	requireBodyNotEmpty := func(rt *RestTester, docID, revID, bodyValue string) {
+		doc, err := rt.GetDatabase().GetDocument(docID, db.DocUnmarshalAll)
+		require.NoError(t, err, "Error reading document from bucket")
+		require.Equal(t, revID, doc.SyncData.CurrentRev)
+		require.NotEmpty(t, doc.Body(), "Document body shouldn't be empty")
+		require.Equal(t, bodyValue, doc.GetDeepMutableBody()["key"])
+	}
+
+	// createOrUpdateDoc creates a new document or update an existing document with the
+	// specified document id, revision id and body value in a channel named "alice".
+	createOrUpdateDoc := func(rt *RestTester, docID, revID, bodyValue string) string {
+		body := fmt.Sprintf(`{"key":%q,"channels":["alice"]}`, bodyValue)
+		dbURL := "/db/" + docID
+		if docID != "" {
+			dbURL = "/db/" + docID + "?rev=" + revID
+		}
+		resp := rt.SendAdminRequest(http.MethodPut, dbURL, body)
+		assertStatus(t, resp, http.StatusCreated)
+		require.NoError(t, rt.WaitForPendingChanges())
+		return respRevID(t, resp)
+	}
+
+	defaultConflictResolverWithTombstoneTests := []struct {
+		// A unique name to identify the unit test.
+		name string
+
+		// bodyValue is used to control the digest value for revision generation.
+		bodyValue string
+
+		// revGenHigh is used to make a revision tie by updating the remote doc.
+		revGenTie bool
+
+		// revGenHigh is used to update the doc on remote to keep revision generation higher.
+		revGenHigh bool
+
+		// tombstoneExpected represents the expected document revision.
+		tombstoneExpected string
+	}{
+		{
+			// Revision tie with local digest is lower than the remote digest.
+			// local generation = remote generation:
+			//	- e.g. local is 3-a(T), remote is 3-b
+			name:              "revGenTieLocalDigestLower",
+			revGenTie:         true,
+			bodyValue:         "EADGBE",
+			tombstoneExpected: "6-6e092cbb01fd955ffa78b518ecfd3eb2",
```

Comment:
The fact that we're resolving a conflict between 3-a and 3-b, and ending up with an expected tombstone with generation 6 should have been a red flag about the error in requiredAdditionalRevs above.  CBG-1049 means that we're going to end up with an additional generation on the local, but that still means the required generation of the winning tombstone should be 5.